src/flask_httpauth.py

```diff
@@ -195,6 +195,49 @@ def ensure_sync(self, f):
             return f
 
 
+class HTTPCookieAuth(HTTPAuth):
+    def __init__(self, scheme=None, realm=None, cookie_name=None):
+        super(HTTPCookieAuth, self).__init__(
+            scheme or 'Bearer',
+            realm,
+            cookie_name
+        )
+
+        self.verify_cookie_callback = None
+        self.cookie_name = cookie_name
+
+    def verify_cookie(self, f):
+        self.verify_cookie_callback = f
+        return f
+
+    def authenticate(self, auth, _):
+        cookie = getattr(auth, 'token', '')
+        if self.verify_cookie_callback:
+            return self.ensure_sync(self.verify_cookie_callback)(cookie)
+
+    def get_auth(self):
+        expected_cookie_name = self.cookie_name or 'Authorization'
```

Comment:
Using a cookie named `Authorization` is confusing. Does anybody do this? I have never seen it.